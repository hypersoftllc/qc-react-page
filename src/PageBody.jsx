import PT from 'prop-types'
import React from 'react'
import Block from '@qc/react-block'


function PageBody(props) {
  const domProps = {
    ...props,
    className: `${props.compClassName} ${props.className}`,
  }
  delete domProps.compClassName
  return (
    <Block {...domProps}>
      {props.children}
    </Block>
  )
}

PageBody.defaultProps = {
  className: '',
  compClassName: 'Page-Body',
  compType: 'div',
}

PageBody.propTypes = {
  children: PT.node,
  className: PT.string,
  compClassName: PT.string,
  compType: PT.oneOf(['article', 'div', 'main', 'section']),
}

export default PageBody
