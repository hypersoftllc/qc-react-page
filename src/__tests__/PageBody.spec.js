/* eslint-env jest */
/* eslint padded-blocks: "off" */

import React from 'react'
import TestRenderer from 'react-test-renderer'

import PageBody from '../PageBody'


describe('PageBody', () => {

  it('should be a function', () => {
    expect(typeof PageBody).toBe('function')
  })

  it('should be a functional component', () => {
    let comp

    comp = TestRenderer.create(
      <PageBody/>
    )

    // Functional components don't have instances.
    expect(comp.getInstance()).toBe(null)
  })

  describe('rendered', () => {

    describe('with no props', () => {

      it('should use default props', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageBody/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with children', () => {

      it('should render children', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageBody>
            <span>Child</span>
          </PageBody>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with className', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageBody className="Foo"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compClassName', () => {

      it('should render with expected class attribute', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageBody compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageBody className="foo" compClassName=""/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageBody className="foo" compClassName="page-body"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

    describe('with compType', () => {

      it('should render with expected component', () => {
        let comp, tree

        comp = TestRenderer.create(
          <PageBody compType="article"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageBody compType="div"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageBody compType="main"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()

        comp = TestRenderer.create(
          <PageBody compType="section"/>
        )

        tree = comp.toJSON()
        expect(tree).toMatchSnapshot()
      })

    })

  })

})
