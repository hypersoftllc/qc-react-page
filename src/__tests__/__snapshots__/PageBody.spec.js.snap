// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBody rendered with children should render children 1`] = `
<div
  className="Block Page-Body"
>
  <span>
    Child
  </span>
</div>
`;

exports[`PageBody rendered with className should render with expected class attribute 1`] = `
<div
  className="Block Page-Body Foo"
/>
`;

exports[`PageBody rendered with compClassName should render with expected class attribute 1`] = `
<div
  className="Block"
/>
`;

exports[`PageBody rendered with compClassName should render with expected class attribute 2`] = `
<div
  className="Block  foo"
/>
`;

exports[`PageBody rendered with compClassName should render with expected class attribute 3`] = `
<div
  className="Block page-body foo"
/>
`;

exports[`PageBody rendered with compType should render with expected component 1`] = `
<article
  className="Block Page-Body"
/>
`;

exports[`PageBody rendered with compType should render with expected component 2`] = `
<div
  className="Block Page-Body"
/>
`;

exports[`PageBody rendered with compType should render with expected component 3`] = `
<main
  className="Block Page-Body"
/>
`;

exports[`PageBody rendered with compType should render with expected component 4`] = `
<section
  className="Block Page-Body"
/>
`;

exports[`PageBody rendered with no props should use default props 1`] = `
<div
  className="Block Page-Body"
/>
`;
